src/components/layout.tsx

```diff
@@ -1,18 +1,54 @@
 import React from 'react';
+import useSWR from 'swr';
 import { Global } from '@emotion/react';
+import getConfig from 'next/config';
+import { useSession } from 'next-auth/react';
 import { UnifiedFooter } from '@mdb/consistent-nav';
 import { globalStyles, Main } from '../styled/layout';
 import { UnifiedNavCustom } from '../styled/consistent-nav';
 import SecondaryNav from './seconardynavnew/';
 
-const Layout: React.FunctionComponent = ({ children }) => {
+interface LayoutProps {
+    pagePath?: string | null;
+}
+
+const Layout: React.FunctionComponent<LayoutProps> = ({
+    children,
+    pagePath,
+}) => {
+    const { data, error } = useSWR('/api/auth/session', async url => {
+        const res = await fetch(url);
+        if (!res.ok) {
+            throw new Error('Failed to load session.');
+        }
+        return res.json();
+    });
+
+    const { data: session } = useSession();
+    const { publicRuntimeConfig } = getConfig();
+    const { absoluteBasePath, accountPortalUrl } = publicRuntimeConfig;
+
+    // For sign in handling, the user will need to be able to return
+    // to the page they were previously on before clicking the login button. The
+    // "fromPagePath" query parameter will hold this value.
+    const fromPagePath = pagePath ? `?fromPagePath=${pagePath}` : '';
+    // The "fromURI" will be sent to the account portal for redirection, and will
+    // include the above information for when it redirects back to DevCenter.
+    const signInParams = new URLSearchParams({
+        fromURI: `${absoluteBasePath}/auth/signin/${fromPagePath}`,
+    });
+    const signInUrl = `${accountPortalUrl}?` + signInParams.toString();
+
     return (
         <>
             <Global styles={globalStyles} />
             <UnifiedNavCustom
                 position="sticky"
                 floraTheme="default"
                 property={{ name: 'DEVHUB', searchParams: [] }}
+                hideTryFree={!!session}
+                hideSignIn={!!session}
```

Comment:
There is currently a FOUC due to some of the pages still being statically rendered for page speed and optimization. As part of [DEVHUB-1290](https://jira.mongodb.org/browse/DEVHUB-1290) I am going to be using `next-plugin-preval` to make all static pages server-side rendered-- this is because we still need to optimize a lot of the queries on the pages such as topic and topic content (if we make these SSR, they will be very slow). However if we use a combination of preval with SSR, we can avoid this issue while still rendering the session server-side.